tests/promisifiedJWTVerify.test.ts

```diff
@@ -0,0 +1,36 @@
+import jwt from "jsonwebtoken";
+
+import { promisifiedJWTVerify } from "../src/utils/promisifiedJWTVerify";
+
+describe("promisifiedJWTVerify", () => {
+  const mockedJWTPayload = {
+    aud: ["yoga-community"],
+    exp: Date.now(),
+    iss: "foo",
+    scope: "phone email",
+    sub: "2a14bdd2-3628-4912-a76e-fd514b5c27a8",
+  };
+
+  const mockedClientSecret = "bar";
+
+  const JWT = jwt.sign(mockedJWTPayload, mockedClientSecret);
+
+  test("it should verify the JWT, and return it decoded", async () => {
+    const decoded = await promisifiedJWTVerify<Record<string, unknown>>({
+      clientSecret: mockedClientSecret,
+      accessToken: JWT,
+    });
+
+    expect(decoded).not.toBe(undefined);
+    expect(decoded).toEqual(expect.objectContaining(mockedJWTPayload));
+  });
+
+  test("it should throw an error is client secret us falsy", async () => {
+    expect.assertions(1);
+
+    await promisifiedJWTVerify<Record<string, unknown>>({
+      clientSecret: undefined,
+      accessToken: JWT,
+    }).catch((error) => expect(error).toBe("Missing client_secret"));
```

Comment:
```suggestion
  expect(() => {
   promisifiedJWTVerify<Record<string, unknown>>({
      clientSecret: undefined,
      accessToken: JWT,
    })
   }).toThrow("Missing client_secret");
```

just to test if it does not work

Reply:
It does not:

```
Expected substring: "Missing client_secret"

    Received function did not throw

      40 |         accessToken: JWT,
      41 |       });
    > 42 |     }).toThrow("Missing client_secret");
         |        ^
      43 |   });
      44 | });
      45 |
```

Reply:
ah yes, found it I think https://jestjs.io/docs/en/tutorial-async#rejects

Reply:
Ah yeah, I `reject` with a string, and not an error. Make sense.

Reply:
Ah nice, but I won't need it if I implement the error from the next comment.